### DISCUSSION TITLE
Possible to append recorded video?

### DISCUSSION THREAD

**Discussion by Taps7734** [bluenviron/mediamtx#3159] [Questions]
### Question

If I have the recording path set to **/recordings/%path/%Y/%m-%d/%Y.%m.%d-%H** so that it splits up the recordings by the hour.  If for some reason the stream is interupted, and restarts, instead of appending the new video to the file, it starts fresh and I lose the beginning.

Is there any way to change this behavior?  Or should I just do this in an external process?

**Comment by aler9** [comment]
Hello, this cannot currently be implemented since a publisher, during disconnections, is free to change the number of tracks or their codecs, while a recording file only supports a fixed list of tracks and codecs.

If recording segments are very close to each other (< 500ms) and contain the same tracks and codecs, you might be able to get a unique recording file by using the [playback server](https://github.com/bluenviron/mediamtx?tab=readme-ov-file#playback-recorded-streams), which is able to merge together recording segments close to each other.

[a comment by an automated account is not shown]
